### PR TITLE
Set network spans with non-successful error codes as failed

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Set the status of a network span based on the http status code

## Testing

Updated unit and integration tests to verify this

### CODE REVIEW

**Comment by bidetofevil** [comment]
* **#962** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/962?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/NetworkRequestApiTest.kt [hunk 1]
```diff
@@ -263,12 +264,20 @@ internal class NetworkRequestApiTest {
                     assertEquals(expectedRequest.bytesReceived.toString(), this.attributes["http.response.body.size"])
                     assertEquals(null, this.attributes["error.type"])
                     assertEquals(null, this.attributes["error.message"])
+                    val statusCode = expectedRequest.responseCode
+                    val expectedStatus = if ( statusCode != null && statusCode >= 200 && statusCode < 400) {
```
Comment (fractalwrench) [inline embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/NetworkRequestApiTest.kt:268]:
```suggestion
                    val expectedStatus = if (statusCode != null && statusCode >= 200 && statusCode < 400) {
```

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/NetworkRequestApiTest.kt (hunk 1 above)
Comment (bidetofevil) [reply embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/NetworkRequestApiTest.kt:268]: Haha can't believe this wasn't caught by lint

**Pull request merged**